### ISSUE TITLE
Adding examples of usage (maybe even migrations from custom/Gdk code)

### ISSUE THREAD

**Issue by rilian-la-te** [sardemff7/libgwater#1]
I want to port X Event driven program from Gdk to pure XCB and GLib

**Comment by sardemff7** [comment]
The basic idea is that GWater XCB is just here to hook XCB events to the GLib main loop.

You create a GSource with `g_water_xcb_source_new` (basically a wrapper around `xcb_connect`) passing your own callback that will handle events.

Then you get the XCB connection using `g_water_xcb_source_get_connection` and start sending your requests with libxcb.

You /should/ have all the needed logic in sardemff7/eventd@8c5ae2a62f2cbb17608e8259f8e25b9d2636b447 (see the `xcb.c` file modified in this commit if the diff is not enough, even though the rest is not really GWater-related).

I modify this issue to a more generic thing so that I will not forget to add basic documentation and example to help people.

Glad to see someone interested in GWater, it is quite surprising!

**Comment by rilian-la-te** [comment]
But there is a problem (but not GWater - related). If app is a gtk app, no
events will be captured.

**Comment by sardemff7** [comment]
Ok, then I will make sure people understand GWater is for GLib-only applications. :-)

**Comment by rilian-la-te** [comment]
This is not GLIb - only:) XCB connection source is GLib - only, but ALSA
source works just fine in GTK app:)

**Comment by beroal** [comment]
I wonder whether your library does what [this feature request for XCB](https://bugs.freedesktop.org/show_bug.cgi?id=29657) describes.

**Comment by sardemff7** [comment]
Not exactly. The request is about adding to the XCB API the proper functions to implement `prepare` and `check`.
My XCB source uses `xcb_poll_for_event` in `check` to fill a queue, then `dispatch` only operates on this queue.

GWater fills my needs and probably other people’s, but I cannot guarantee it is doing the right thing. :-)
